Docs improvements (#961)
anatawa12/AvatarOptimizer: .docs/content/docs/faq/index.ja.md

Reviewer: T&Oの項目を上の方に上げません? (enも同様)

Author: 忘れてた。やります

Author: fd673c2e530df9030896095e3e15fa56e2d6daf3

---
title: よくある質問
weight: 2
---

# よくある質問 {#faq}

AvatarOptimizerに関するよくある質問のリストです。
他に質問がある場合は、[GitHub Discussions]または[fediverse]でお気軽にお尋ねください。

## メッシュが視界の中にあるのに非表示になってしまう {#mesh-is-invisible-even-though-it-is-in-the-field-of-view}

> メッシュが視界の中にあるのに非表示になってしまう問題は、多くの場合、Skinned Mesh RendererのRoot Boneの指定しわすれである事が多いです。

Reviewer: ```suggestion
多くの場合、この問題はSkinned Mesh Rendererの`Root Bone`を指定し忘れていることが原因です。
```

> [`MergeSkinnedMesh`]はRoot Boneを自動的に設定しないため、MergeSkinnedMesh用のSkinned Mesh RendererのRoot Boneも手動で設定する必要があります。

Reviewer: ```suggestion
[`AAO Merge Skinned Mesh`]コンポーネントは`Root Bone`の設定を自動的には行わないため、統合先のメッシュの`Root Bone`を手動で設定する必要があります。
(統合先のメッシュとは、`AAO Merge Skinned Mesh`コンポーネントと一緒に付いているSkinned Mesh Rendererのことです。)
```

Author: ```suggestion
[`AAO Merge Skinned Mesh`]コンポーネントは`Root Bone`の設定を自動的には行わないため、統合先のメッシュ[^1]の`Root Bone`を手動で設定する必要があります。
```

> もし[Modular Avatar]を使用している場合には、アバターのルートに[`MA Mesh Settings`]を追加して、Root Boneを設定することでアタバー全体のRoot BoneとBoundsを設定することができます。

Reviewer: ```suggestion
[Modular Avatar]を使用している場合は、アバターのルートに[`MA Mesh Settings`]コンポーネントを追加して設定することにより、アバター全体の`Root Bone`と`Bounds`を設定することができます。
```

> ## `MergeSkinnedMesh`で結合したメッシュの明るさが他のメッシュと異なる {#the-brightness-of-the-meshes-merged-with-mergeskinnedmesh-is-different-from-other-meshes}

Reviewer: ```suggestion
## `AAO Merge Skinned Mesh`コンポーネントで統合したメッシュの明るさが他のメッシュと異なる {#the-brightness-of-the-meshes-merged-with-aao-merge-skinned-mesh-component-is-different-from-other-meshes}
```

> [`MergeSkinnedMesh`]で結合したメッシュの明るさが他のメッシュと異なる問題は、多くの場合、結合したメッシュのAnchor Overrideの指定し忘れが原因です。

Reviewer: ```suggestion
多くの場合、この問題は統合先のメッシュの`Anchor Override`を指定し忘れていることが原因です。
```

> [`MergeSkinnedMesh`]はAnchor Overrideを自動的に設定しないため、MergeSkinnedMesh用のSkinned Mesh RendererのAnchor Overrideも手動で設定する必要があります。

Reviewer: ```suggestion
[`AAO Merge Skinned Mesh`]コンポーネントは`Anchor Override`の設定を自動的には行わないため、統合先のメッシュの`Anchor Override`を手動で設定する必要があります。
(統合先のメッシュとは、`AAO Merge Skinned Mesh`コンポーネントと一緒に付いているSkinned Mesh Rendererのことです。)
```

Author: ```suggestion
[`AAO Merge Skinned Mesh`]コンポーネントは`Anchor Override`の設定を自動的には行わないため、統合先のメッシュ[^merged-mesh]の`Anchor Override`を手動で設定する必要があります。
```

> もし[Modular Avatar]を使用している場合には、アバターのルートに[`MA Mesh Settings`]を追加して、Anchor Overrideを設定することでアタバー全体のAnchor Overrideを設定することができます。

Reviewer: ```suggestion
[Modular Avatar]を使用している場合は、アバターのルートに[`MA Mesh Settings`]コンポーネントを追加して設定することにより、アバター全体の`Anchor Override`を設定することができます。
```

> ## `MergeSkinnedMesh`を使用するとマテリアルのアニメーションが競合する {#material-animations-conflict-when-using-mergeskinnedmesh}

Reviewer: ```suggestion
## `AAO Merge Skinned Mesh`コンポーネントを使用すると、マテリアルに対するアニメーションが競合する {#material-animations-conflict-when-using-aao-merge-skinned-mesh-component}
```

> [`MergeSkinnedMesh`]を使用するとマテリアルのアニメーションが競合する問題は、既知のバグで、現在のところは競合するのが仕様です。

Reviewer: ```suggestion
この問題は既知のバグであり、現時点では競合してしまう仕様です。
```

> マテリアルのアニメーションを行っているメッシュを結合する際には、競合しないように気をつけて結合してください。

Reviewer: ```suggestion
マテリアルに対してアニメーションされるメッシュを統合する際は、競合しないようご注意ください。
```

> 競合する場合には警告が表示されるので、警告を確認してください。

Reviewer: ```suggestion
競合する場合は警告が表示されるので、そちらを確認してください。
```

この問題のissue: [#340](https://github.com/anatawa12/AvatarOptimizer/issues/340)

> ## `MergeSkinnedMesh`を使用するとBlendShapeのアニメーションが競合する {#blendshape-animations-conflict-when-using-mergeskinnedmesh}

Reviewer: ```suggestion
## `AAO Merge Skinned Mesh`コンポーネントを使用すると、BlendShapeに対するアニメーションが競合する {#blendshape-animations-conflict-when-using-aao-merge-skinned-mesh-component}
```

> [`MergeSkinnedMesh`]を使用するとBlendShapeのアニメーションが競合する問題は、既知のバグで、現在のところは競合するのが仕様です。

Reviewer: ```suggestion
この問題は既知のバグであり、現時点では競合してしまう仕様です。
```

> BlendShapeのアニメーションを行っているメッシュを結合する際には、競合しないように気をつけて結合してください。

Reviewer: ```suggestion
BlendShapeに対してアニメーションされるメッシュを統合する際は、競合しないようご注意ください。
```

> 競合する場合には警告が表示されるので、警告を確認してください。

Reviewer: ```suggestion
競合する場合は警告が表示されるので、そちらを確認してください。
```

この問題のissue: [#568](https://github.com/anatawa12/AvatarOptimizer/issues/568)

> ## `Trace and Optimize`を使用するとアバターの振る舞いが変わる {#avatar-behavior-changed-when-using-trace-and-optimize}

Reviewer: ```suggestion
## `AAO Trace and Optimize`コンポーネントを使用すると、アバターの振る舞いや見た目が変わる {#avatar-behavior-changed-when-using-aao-trace-and-optimize-component}
```

> `Trace and Optimize`を使用するとアバターの振る舞いが変わる場合は、AvatarOptimizerの他のコンポーネントのバグに依存していない限り、バグです。

Reviewer: ```suggestion
`AAO Trace and Optimize`コンポーネントを使用して、アバターの振る舞いや見た目が変わってしまった場合は、(アバターがAvatarOptimizerの他のコンポーネントのバグ挙動に依存していない限り、)全てバグです。例外はありません。
```

Author: ()に例外が書かれてて、その後ろに"例外はありません"はなんか違う気がする

Reviewer: 確かに(何も考えてなかった)

Reviewer: 消しました

> [GitHub issues], [misskey][fediverse] (mastodon), [twitter] などで報告してください。

Reviewer: ```suggestion
[GitHub issues]や[misskey][fediverse] (mastodon)、[twitter]などから報告をお願いいたします。
```

Author: "報告をお願いいたします。"、もう少しナチュラルな"報告をお願いします。"くらいでもいいとは思ったりしたけどまぁお願いだから"報告をお願いいたします。"も妥当だしちょっとどうしようってなった

Reviewer: 悩むけど、カジュアルでも良いかもです

## Avatar Optimizerの開発を支援したい {#i-want-to-support-the-development-of-avatar-optimizer}

> Avatar Optimizerの開発を支援したい場合は、[GitHub Discussions]でのフィードバックや[GitHub issues]でのバグ報告、機能リクエスト、プルリクエスト等を歓迎します。

Reviewer: ```suggestion
Avatar Optimizerの開発を支援したい場合、プルリクエストや[GitHub Discussions]でのフィードバック、[GitHub issues]でのバグ報告や機能追加要望などを歓迎しています。
```
順番変わっちゃいましたが、なんかそのままの順番だと弄っても日本語がしっくり来なかったので…

Reviewer: 本当はPRは重いので後ろに置きたいけど、github issuesに繋がって見えてしまう

Author: ```suggestion
Avatar Optimizerの開発を支援したい場合、[GitHub Discussions]でのフィードバックや[GitHub issues]でのバグ報告、機能リクエスト、またプルリクエストなどを歓迎しています。
```
とか？

Reviewer: アリかも

Reviewer: "や"だけ欲しいね

Reviewer: ```suggestion
Avatar Optimizerの開発を支援したい場合、[GitHub Discussions]でのフィードバックや[GitHub issues]でのバグ報告、機能追加の要望、またプルリクエストなどを歓迎しています。
```

> [good first issue]がついたissueは比較的実装が簡単なissueです。初めてのプルリクエストにおすすめです。

Reviewer: ```suggestion
[good first issue]が付いているissueは、比較的実装が簡単なissueです。初めてのプルリクエストにおすすめです。
```
```suggestion
[good first issue]が付いているissueは、比較的実装が簡単なissueです。初めてのコントリビューションにおすすめです。
```

> また、[help wanted]がついたissueは開発者や情報などが足りないissueです。議論や開発に参加していただけると助かります。

Reviewer: ```suggestion
また、[help wanted]が付いているissueは、開発者や情報などが不足しているissueです。議論や開発に参加していただけると助かります。
```

> [github sponsors] や [booth] で金銭的な支援も受け付けています。

Reviewer: ```suggestion
なお、[github sponsors]や[booth]での金銭的な支援も受け付けています。
```

[fediverse]: https://misskey.niri.la/@anatawa12
[GitHub Discussions]: https://github.com/anatawa12/AvatarOptimizer/discussions
[GitHub issues]: https://github.com/anatawa12/AvatarOptimizer/issues/new/choose

> [`MergeSkinnedMesh`]: ../reference/merge-skinned-mesh/

Reviewer: ```suggestion
[`AAO Merge Skinned Mesh`]: ../reference/merge-skinned-mesh/
```

[Modular Avatar]: https://modular-avatar.nadena.dev/
[`MA Mesh Settings`]: https://modular-avatar.nadena.dev/docs/reference/mesh-settings
[twitter]: https://twitter.com/anatawa12_vrc
[github sponsors]: https://github.com/sponsors/anatawa12
[booth]: https://anatawa12.booth.pm/items/4885109
[good first issue]: https://github.com/anatawa12/AvatarOptimizer/labels/good%20first%20issue

> [help wanted]: https://github.com/anatawa12/AvatarOptimizer/labels/help%20wanted

Author: ```suggestion
[help wanted]: https://github.com/anatawa12/AvatarOptimizer/labels/help%20wanted

[^merged-mesh]: 統合先のメッシュとは、`AAO Merge Skinned Mesh`コンポーネントと一緒に付いているSkinned Mesh Rendererのことです。
```

---
title: よくある質問
weight: 2
---

# よくある質問 {#faq}

AvatarOptimizerに関するよくある質問のリストです。
他に質問がある場合は、[GitHub Discussions]または[fediverse]でお気軽にお尋ねください。

## メッシュが視界の中にあるのに非表示になってしまう {#mesh-is-invisible-even-though-it-is-in-the-field-of-view}

多くの場合、この問題はSkinned Mesh Rendererの`Root Bone`を指定し忘れていることが原因です。

[`AAO Merge Skinned Mesh`]コンポーネントは`Root Bone`の設定を自動的には行わないため、統合先のメッシュ[^merged-mesh]の`Root Bone`を手動で設定する必要があります。

[Modular Avatar]を使用している場合は、アバターのルートに[`MA Mesh Settings`]コンポーネントを追加して設定することにより、アバター全体の`Root Bone`と`Bounds`を設定することができます。

## `AAO Merge Skinned Mesh`コンポーネントで統合したメッシュの明るさが他のメッシュと異なる {#the-brightness-of-the-meshes-merged-with-aao-merge-skinned-mesh-component-is-different-from-other-meshes}

多くの場合、この問題は統合先のメッシュの`Anchor Override`を指定し忘れていることが原因です。

[`AAO Merge Skinned Mesh`]コンポーネントは`Anchor Override`の設定を自動的には行わないため、統合先のメッシュ[^merged-mesh]の`Anchor Override`を手動で設定する必要があります。

[Modular Avatar]を使用している場合は、アバターのルートに[`MA Mesh Settings`]コンポーネントを追加して設定することにより、アバター全体の`Anchor Override`を設定することができます。

> ## `AAO Merge Skinned Mesh`コンポーネントを使用すると、マテリアルプロパティに対するアニメーションが競合する {#material-animations-conflict-when-using-aao-merge-skinned-mesh-component}

Reviewer: ```suggestion
## `AAO Merge Skinned Mesh`コンポーネントを使用すると、マテリアルプロパティに対するアニメーションが競合する {#material-property-animations-conflict-when-using-aao-merge-skinned-mesh-component}
```

この問題は既知のバグであり、現時点では競合してしまう仕様です。
マテリアルプロパティに対してアニメーションされるメッシュを統合する際は、競合しないようご注意ください。
競合する場合は警告が表示されるので、そちらを確認してください。

この問題のissue: [#340](https://github.com/anatawa12/AvatarOptimizer/issues/340)

## `AAO Merge Skinned Mesh`コンポーネントを使用すると、マテリアルスロットに対するアニメーションが競合する {#material-slot-animations-conflict-when-using-aao-merge-skinned-mesh-component}

> `AAO Merge Skinned Mesh`コンポーネントは、デフォルトで同じマテリアルを持つメッシュのマテリアルスロットを統合します。

Reviewer: ```suggestion
`AAO Merge Skinned Mesh`コンポーネントは、統合対象のメッシュで同じマテリアルを使用しているマテリアルスロットをデフォルトで統合します。
```
mergeSMRのdocsに寄せ

> これにはアニメーションされているマテリアルスロットも含まれます。

Reviewer: ```suggestion
これにはアニメーションされるマテリアルスロットも含まれます。
```

> アニメーションによってマテリアルが異なるように置き換えられる場合は、`AAO Merge Skinned Mesh`コンポーネントの`Merge Materials`の`Merge`をオフにしてください。

Reviewer: ```suggestion
アニメーションによってそれぞれのマテリアルを別のものに置き換える場合は、`AAO Merge Skinned Mesh`コンポーネントの`マテリアルの統合`の`統合する`をオフにしてください。
```
i18n

`AAO Merge Skinned Mesh`コンポーネントは、統合対象のメッシュで同じマテリアルを使用しているマテリアルスロットをデフォルトで統合します。
これにはアニメーションされるマテリアルスロットも含まれます。

アニメーションによってそれぞれのマテリアルを別のものに置き換える場合は、`AAO Merge Skinned Mesh`コンポーネントの`マテリアルの統合`の`統合する`をオフにしてください。

## `AAO Merge Skinned Mesh`コンポーネントを使用すると、BlendShapeに対するアニメーションが競合する {#blendshape-animations-conflict-when-using-aao-merge-skinned-mesh-component}

> この問題は既知のバグであり、現時点では競合してしまう仕様です。

Reviewer: ```suggestion
この問題は既知のバグであり、現時点では競合してしまう仕様です。

```

---
title: よくある質問
weight: 2
---

# よくある質問 {#faq}

Avatar Optimizerに関するよくある質問のリストです。
他に質問がある場合は、[GitHub Discussions]または[Fediverse]でお気軽にお尋ねください。

> ## `AAO Trace and Optimize`コンポーネントを使用すると、アバターの振る舞いや見た目が変わる {#avatar-behavior-changed-when-using-aao-trace-and-optimize-component}

Author: ```suggestion
## `AAO Trace and Optimize`コンポーネントを使用すると、アバターの振る舞いや見た目が変わる {#avatar-behavior-or-appearance-changed-when-using-aao-trace-and-optimize-component}
```
